### ISSUE TITLE
Trailing comma added incorrectly in TypeScript type definition

### ISSUE THREAD

**Issue by zxbodya** [prettier/prettier#7834]
**Prettier 2.0.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEcAeAHCAnGACGATwzjwEE8BePYAHSjzwEMB+JG+xxgbU68bTsAzjGwBLKAHM+jALrsiJCADM8AZQgBbOABlokvVMOSAKsTgA5Jtu4BrOIRUFzTjduMf9Zkle2yA3HwAvngAPnhQAK4ANtGBUEH+IAA0IBAYMGLQQsigTNjYEADuAAr5CDkoTNFFTIQ5qQBG2Exg9jBqGK0SksiikXCpABYwmtEA6kNi8EJdYHBqFdNiAG7ThMjgQg0gEkJwuCUtkppMyMrV+6kAVkJoAEItbXAd1roScOeXgyC3aGo9aJwACKkQg8C+0SuIC62H22E2ijgQjA4gyKRh4lg4zEABMYENkAAOAAMqQwhX24xaGE2FORBxWn1SAEcwfAjulKiAmEIALRQOBwXHCjHYOBssTio5ME5nJAXKE-faaMR9bADVJCQEg9mfBXfVIwJiNHH4wlIABMRpaYmiPQAwlpTptkQBWDGRfYmE2VRXQlYDACSUBFsDUqLEGTIobURCBkP2QSCQA)
```sh
--parser typescript
```

**Input:**
```tsx
export type A = {
  a?: {
    [
      x: string
    ]: typeof SomeLongLongLongTypeName[keyof typeof SomeLongLongLongTypeName];
  } | null;
};
```

**Output:**
```tsx
export type A = {
  a?: {
    [
      x: string,
    ]: typeof SomeLongLongLongTypeName[keyof typeof SomeLongLongLongTypeName];
  } | null;
};

```

**Expected behavior:**

Keep input as-is - comma should not be added after `x: string` because it is invalid syntax.

**Comment by thorn0** [comment]
While I agree that this comma makes little sense, are you sure this is invalid syntax?

**Comment by zxbodya** [comment]
Actually you right it is a valid syntax for TypeScript  - [TypeScript Playground Link](https://www.typescriptlang.org/play?#code/DYUwLgBAyg9gtiAMjAdgc2ezaAqBPABxADkBDBCAXggG8BfCUgZ1tIC4UBXOAIxACc6AbgBQIkAA8CMfpDCEQEAIJVaIiIwD8bNRo0BtdXo0SdTMPwCW6ADRGNAXR3yiMAGbR4SVBh-Z8RGQI+gDWIHjuEC4gkbAI2Ak+ASTkIA6iGgwAPhBcwMCiwiJAA)

But, for Babel it is not(where I caught it), it is not: [Babel Playground](https://babeljs.io/repl#?browsers=ie%209&build=&builtIns=usage&spec=false&loose=false&code_lz=DYUwLgBAyg9gtiAMjAdgc2ezaAqBPABxADkBDBCAXggG8BfCUgZ1tIC4UBXOAIxACc6AbgBQIkAA8CMfpDCEQEAIJVaIiIwD8bNRo0BtdXo0SdTMPwCW6ADRGNAXR3yiMAGbR4SVBh_Z8RGQI-gDWIHjuEC4gkbAI2Ak-ASTkIA6iGgwAPhBcwMCiwiJAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Ces2015%2Ces2016%2Ces2017%2Creact%2Cstage-0%2Cstage-1%2Ctypescript%2Cenv&prettier=false&targets=Node-8.9&version=7.9.0&externalPlugins=)

**Comment by bakkot** [comment]
I'll submit a PR for this. I think the right behavior is to omit the trailing comma when there's just one item.

**Comment by thorn0** [comment]
I guess we'll release the fix as v2.0.2 tomorrow.
